workers/event_catcher/event_catcher.rb

```diff
@@ -0,0 +1,187 @@
+require "manageiq-messaging"
+require "sd_notify"
+require "rbvmomi"
+
+class EventCatcher
+  def initialize(ems_id, default_endpoint, default_authentication, messaging_opts, page_size = 20)
+    @ems_id         = ems_id
+    @hostname       = default_endpoint["hostname"]
+    @username       = default_authentication["userid"]
+    @password       = default_authentication["password"]
+    @port           = default_endpoint["port"]
+    @messaging_host = messaging_opts["host"]
+    @messaging_port = messaging_opts["port"]
+    @page_size      = page_size
+  end
+
+  def run!
+    vim                     = connect
+    event_history_collector = create_event_history_collector(vim, page_size)
+    property_filter         = create_property_filter(vim, event_history_collector)
+
+    notify_started
+
+    wait_for_updates(vim) do |property_change|
+      next unless property_change.name.match?(/latestPage.*/)
```

Comment:
Super minor, but from a readability perspective, we might want to move this into a `filtered_event?` method or something like that.

Reply:
Unless this means something else?

Reply:
This isn't about specific event filtering, to be honest I just copied this logic from https://github.com/ManageIQ/vmware_web_service/blob/master/lib/VMwareWebService/MiqVimEventMonitor.rb#L66  Without fully understanding why it is there I didn't want to delete it.

The PropertyFilterSpec path specifies just the latestPage [[ref]](https://github.com/ManageIQ/vmware_web_service/blob/master/lib/VMwareWebService/MiqVimEventMonitor.rb#L33) so this might not be needed, maybe it was there to only process this property_collector in case there were others but usually that's done by checking the property filter ref.

Reply:
I would not bring in code that you cannot understand.  Maybe in the past, there was no way (or we did not know how) to do the latestPage filtering via the PropertyFilterSpec.